front/pages/api/w/[wId]/data_sources/managed.ts

```diff
@@ -104,6 +108,39 @@ async function handler(
           },
         });
       }
+      switch (provider) {
+        case "webcrawler": {
+          if (!auth.isBuilder()) {
+            return apiError(req, res, {
+              status_code: 403,
+              api_error: {
+                type: "data_source_auth_error",
+                message:
+                  "Only the users that are `builders` for the current workspace can add a public website.",
+              },
+            });
+          }
+          break;
+        }
+        case "confluence":
+        case "github":
+        case "google_drive":
+        case "intercom":
+        case "notion":
+        case "slack": {
+          if (!auth.isAdmin()) {
+            return apiError(req, res, {
+              status_code: 403,
+              api_error: {
+                type: "data_source_auth_error",
+                message:
+                  "Only the users that are `admins` for the current workspace can create a managed data source.",
+              },
+            });
+          }
+          break;
+        }
```

Comment:
No ok to ship unless we have an assertNever otherwise we won't check future connectors for admin

Reply:
oopsy the whole point of this switch was to add the assertNever. Thanks.